packages/ra-supabase/src/AdminGuesser.tsx

```diff
@@ -0,0 +1,169 @@
+        instanceUrl && apiKey ? createClient(instanceUrl, apiKey) : null;
+    const defaultDataProvider =
+        instanceUrl && apiKey && defaultSupabaseClient
+            ? supabaseDataProvider({
+                  instanceUrl,
+                  apiKey,
+                  supabaseClient: defaultSupabaseClient,
+              })
+            : undefined;
+    const defaultAuthProvider =
+        instanceUrl && apiKey && defaultSupabaseClient
+            ? supabaseAuthProvider(defaultSupabaseClient, {})
+            : undefined;
+
+    return (
+        <BrowserRouter>
+            <AdminContext
+                authProvider={authProvider ?? defaultAuthProvider}
+                basename={basename}
+                darkTheme={darkTheme}
+                dataProvider={dataProvider ?? defaultDataProvider}
+                defaultTheme={defaultTheme}
+                i18nProvider={i18nProvider}
+                lightTheme={lightTheme}
+                queryClient={queryClient}
+                store={store}
+                theme={theme}
+            >
+                <AdminUIGuesser {...rest} />
+            </AdminContext>
+        </BrowserRouter>
+    );
+};
+
+const AdminUIGuesser = (props: AdminUIProps) => {
+    const resourceDefinitions = useCrudGuesser();
+    const { children, ...rest } = props;
+    // while we're guessing, we don't want to show the not found page
+    const [CatchAll, setCatchAll] = React.useState<
+        React.ComponentType | undefined
+    >(() => Loading);
+    React.useEffect(() => {
+        if (!children && resourceDefinitions.length > 0) {
+            console.log(
+                `Guessed Admin:
+
+import { Admin, Resource, CustomRoutes } from 'react-admin';
+import { BrowserRouter, Route } from 'react-router-dom';
+import { createClient } from '@supabase/supabase-js';
+import {
+    CreateGuesser,
+    EditGuesser,
+    ForgotPasswordPage,
+    ListGuesser,
+    LoginPage,
+    SetPasswordPage,
+    ShowGuesser,
+    defaultI18nProvider,
+    supabaseDataProvider,
+    supabaseAuthProvider
+} from 'ra-supabase';   
+
+const instanceUrl = YOUR_SUPABASE_URL;
+const apiKey = YOUR_SUPABASE_API_KEY;
+const supabaseClient = createClient(instanceUrl, apiKey);
+const dataProvider = supabaseDataProvider({ instanceUrl, apiKey, supabaseClient });
+const authProvider = supabaseAuthProvider(supabaseClient, {});
+
+export const App = () => (
+    <BrowserRouter>
+        <Admin
+            dataProvider={dataProvider}
+            authProvider={authProvider}
+            i18nProvider={defaultI18nProvider}
+            loginPage={LoginPage}
+        >${resourceDefinitions
+            .map(
+                def => `
+            <Resource name="${def.name}"${
+                    def.list ? ' list={ListGuesser}' : ''
+                }${def.edit ? ' edit={EditGuesser}' : ''}${
+                    def.create ? ' create={CreateGuesser}' : ''
+                }${def.show ? ' show={ShowGuesser}' : ''} />`
```

Comment:
question: **Contact** and **Sale** default `recordRepresentation` are wrong. Do you think it would be possible to use the OpenAPI schema to fix them?

Reply:
I don't think the OpenAPI spec allows for that. I've seen nothing related in the default PostgREST OpenAPI output, nor in [the PostgREST OpenAPI doc](https://docs.postgrest.org/en/v12/references/api/openapi.html).